### PR TITLE
CY-3114 Replace-certificates system test

### PR DESCRIPTION
**Pull request by YoniItzhak**

### CODE REVIEW

File: cosmo_tester/test_suites/cluster/full_cluster_test.py [hunk 1]
```diff
@@ -221,6 +222,56 @@ def test_workflow_resume_manager_failover(minimal_cluster,
...
+    script_name = 'create_replace_certs_config_script.py'
+    remote_script_path = '/tmp/' + script_name
+    remote_ssh_key_path = '~/.ssh/ssh_key.pem'
+
+    manager.put_remote_file(remote_ssh_key_path, local_ssh_key_path)
+    manager.run_command('cfy profiles set --ssh-user {0} --ssh-key {1}'.format(
+        manager.username, remote_ssh_key_path))
+
+    manager.put_remote_file(remote_script_path,
+                            join(dirname(__file__), script_name))
```
Comment (YoniItzhak) [inline cosmo_tester/test_suites/cluster/full_cluster_test.py:269]: The `dirname(__file__)` might not be needed here?

File: cosmo_tester/test_suites/image_based_tests/aio_replace_certs_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    script_name = 'aio_create_replace_certs_config_script.py'
+    remote_script_path = '/tmp/' + script_name
+    remote_ssh_key_path = '~/.ssh/ssh_key.pem'
+
+    manager.put_remote_file(remote_ssh_key_path, local_ssh_key_path)
+    manager.run_command('cfy profiles set --ssh-user {0} --ssh-key {1}'.format(
+        manager.username, remote_ssh_key_path))
+
+    manager.put_remote_file(remote_script_path,
+                            join(dirname(__file__), script_name))
```
Comment (YoniItzhak) [inline cosmo_tester/test_suites/image_based_tests/aio_replace_certs_test.py:45]: Also here?

File: cosmo_tester/test_suites/cluster/full_cluster_test.py [hunk 3]
```diff
@@ -221,6 +222,56 @@ def test_workflow_resume_manager_failover(minimal_cluster,
...
+                         ' -s {0},{1}'.format(host.private_ip_address,
+                                              host.ip_address))
+        mgr1.run_command('chmod 444 {0}'.format(key_path), use_sudo=True)
+    replace_certs_config_path = '~/certificates_replacement_config.yaml'
+    _create_replace_certs_config_file(mgr1, replace_certs_config_path,
+                                      ssh_key.private_key_path)
+
+    local_new_ca_path = join(str(module_tmpdir), 'new_ca.crt')
+    mgr1.get_remote_file('~/.cloudify-test-ca/ca.crt', local_new_ca_path)
+    mgr1.client._client.cert = local_new_ca_path
```
Comment (YoniItzhak) [inline cosmo_tester/test_suites/cluster/full_cluster_test.py:248]: We only update the client "locally" during the replace_certificates procedure (as seen [here](https://github.com/cloudify-cosmo/cloudify-cli/blob/5e2365dedce24daf3ea5bed445a4f450efe7d75b/cloudify_cli/replace_certificates_config.py#L178)), so we need to update it "again".

**Comment by YoniItzhak** [comment]
The code can be reduced due to code multiplication, but I didn't find a place to put the shared code at. Moreover, I'm not sure if the scripts' location should be as it is now, or at the "resources" directory.

File: cosmo_tester/test_suites/cluster/create_replace_certs_config_script.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import argparse
```
Comment (geokala) [inline cosmo_tester/test_suites/cluster/create_replace_certs_config_script.py:1]: This should probably be somewhere in the resources tree (cosmo_tester/resources/<something>)? 
It doesn't look like it's actually directly invoked test code, but instead a supporting script that is used elsewhere?

File: cosmo_tester/test_suites/cluster/full_cluster_test.py (hunk 1 above)
Comment (geokala) [reply cosmo_tester/test_suites/cluster/full_cluster_test.py:269]: See comment at start of the replace certs script- this should be being uploaded from somewhere under the resources subdir.

File: cosmo_tester/test_suites/image_based_tests/aio_create_replace_certs_config_script.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import argparse
```
Comment (geokala) [inline cosmo_tester/test_suites/image_based_tests/aio_create_replace_certs_config_script.py:1]: Can you avoid the duplication for this by merging it with the other one and having a flag (if needed) to control which type of config it outputs?

**Review by geokala — COMMENTED** [review]
See comments.

File: cosmo_tester/test_suites/cluster/create_replace_certs_config_script.py (hunk 4 above)
Comment (YoniItzhak) [reply cosmo_tester/test_suites/cluster/create_replace_certs_config_script.py]: Correct, moved to be under resources/scripts.

File: cosmo_tester/test_suites/image_based_tests/aio_create_replace_certs_config_script.py (hunk 5 above)
Comment (YoniItzhak) [reply cosmo_tester/test_suites/image_based_tests/aio_create_replace_certs_config_script.py]: Done.

**Review by geokala — APPROVED** [review]

**Pull request merged**